### ISSUE TITLE
How to make n_layer=12 work for Char-Shakespeare?

### ISSUE THREAD

**Issue by flyman3046** [karpathy/nanoGPT#211]
I started with n_layer=6 with a small dataset (100 batches, batch_size=64) and its loss gradually decreased from 3.x to 1.x. It is consistent over a few runs.

But I set n_layer=12, its loss stayed around 3.1, even I ran 100 epochs. Has anyone successfully trained n_layer=12 for Char-Shakespeare? Can you share your config parameters? Thanks.

**Comment by dkobak** [comment]
I cannot reproduce it. I can train the model with `n_layer=12` on `shakespeare_char` with all default parameters, and the loss does go down:
```
250 iterations -- validation loss 2.11
500 iterations -- validation loss 1.77
750 iterations -- validation loss 1.59
1000 iterations -- validation loss 1.52
1250 iterations -- validation loss 1.49
1500 iterations -- validation loss 1.51
1750 iterations -- validation loss 1.52
2000 iterations -- validation loss 1.56
```

For comparison, using default `n_layer=6` I get min val loss 1.48 after 1750 iterations.

**Comment by flyman3046** [comment]
Thanks Dmitry for the verification. I will revisit my script.
